packages/aws-cdk/lib/api/cloudformation-deployments.ts

```diff
@@ -251,6 +268,89 @@ export class CloudFormationDeployments {
     return stack.exists;
   }
 
+  /**
+   * Read a version from an SSM parameter, cached
+   */
+  protected async versionFromSsmParameter(sdk: ISDK, parameterName: string): Promise<number> {
+    const existing = this.ssmCache.get(parameterName);
+    if (existing !== undefined) { return existing; }
+
+    const ssm = sdk.ssm();
+
+    try {
+      const result = await ssm.getParameter({ Name: parameterName }).promise();
+
+      const asNumber = parseInt(`${result.Parameter?.Value}`, 10);
+      if (isNaN(asNumber)) {
+        throw new Error(`SSM parameter ${parameterName} not a number: ${result.Parameter?.Value}`);
+      }
+
+      this.ssmCache.set(parameterName, asNumber);
+      return asNumber;
+    } catch (e) {
+      if (e.code === 'ParameterNotFound') {
+        throw new Error(`SSM parameter ${parameterName} not found. Has the environment been bootstrapped? Please run \'cdk bootstrap\' (see https://docs.aws.amazon.com/cdk/latest/guide/bootstrapping.html)`);
+      }
+      throw e;
+    }
+  }
+
+  /**
+   * Try to use the bootstrap lookupRole. There are two scenarios that are handled here
+   *  1. The lookup role may not exist (it was added in bootstrap stack version 7)
+   *  2. The lookup role may not have the correct permissions (ReadOnlyAccess was added in
```

Comment:
Sheesh, fortunately at version 7 it already has the required SSM parameter permissions... 😅